spectral_cube/stokes_spectral_cube.py

```diff
@@ -170,6 +197,61 @@ def _new_cube_with(self, stokes_data=None,
+            errmsg = "Transformation of a subset of Stokes axes is not yet supported."
+            errmsg_template = "Transformation from {} to {} is not yet supported."
+
+            raise NotImplementedError(errmsg)
+        elif stokes_basis == "Generic":
+            data =  self._stokes_data
+        elif self.stokes_type == "Linear":
+            if stokes_basis == "Circular":
+                errmsg = errmsg_template.format("Linear", "Circular")
+                raise NotImplementedError(errmsg)
+            if stokes_basis == "Sky":
+                data = dict(I = self.__class__(self._stokes_data['XX'] + self._stokes_data['YY'], wcs = self.wcs, mask=self._mask['XX']&self._mask['YY']),
+                            Q = self.__class__(self._stokes_data['XX'] - self._stokes_data['YY'], wcs = self.wcs, mask=self._mask['XX']&self._mask['YY']),
+                            U = self.__class__(self._stokes_data['XY'] + self._stokes_data['YX'], wcs = self.wcs, mask=self._mask['XY']&self._mask['YX']),
+                            V = self.__class__(1j*(self._stokes_data['XY'] - self._stokes_data['YX']), wcs = self.wcs, mask=self._mask['XY']&self._mask['YX']))
+            if stokes_basis == "Linear":
+                data = self._stokes_data
+
+        elif self.stokes_type == "Circular":
+            if stokes_basis == "Linear":
+                errmsg = errmsg_template.format("Circular", "Linear")
+                raise NotImplementedError(errmsg)
+            if stokes_basis == "Sky":
+                data = dict(I = self.__class__(self._stokes_data['RR'] + self._stokes_data['LL'], wcs = self.wcs, mask=self._mask['RR']&self._mask['LL']),
+                            Q = self.__class__(self._stokes_data['RL'] + self._stokes_data['RL'], wcs = self.wcs, mask=self._mask['RL']&self._mask['LR']),
+                            U = self.__class__(1j*(self._stokes_data['RL'] - self._stokes_data['LR']), wcs = self.wcs, mask=self._mask['RL']&self._mask['LR']),
+                            V = self.__class__(self._stokes_data['RR'] - self._stokes_data['LL'], wcs = self.wcs, mask=self._mask['RR']&self._mask['LL']))
+            if stokes_basis == "Circular":
+                data = self._stokes_data
+
+        elif self.stokes_type == "Sky":
+            if stokes_basis == "Linear":
+                data = dict(XX = self.__class__(0.5*(self._stokes_data['I'] + self._stokes_data['Q']), wcs = self.wcs, mask=self._mask['I']&self._mask['Q']),
+                            XY = self.__class__(0.5*(self._stokes_data['U'] + 1j*self._stokes_data['V']), wcs = self.wcs, mask=self._mask['U']&self._mask['V']),
+                            YX = self.__class__(0.5*(self._stokes_data['U'] - 1j*self._stokes_data['V']), wcs = self.wcs, mask=self._mask['U']&self._mask['V']),
+                            YY = self.__class__(0.5*(self._stokes_data['I'] - self._stokes_data['Q']), wcs = self.wcs, mask=self._mask['I']&self._mask['Q']))
+            if stokes_basis == "Sky":
```

Comment:
should be `if stokes_bases == 'Circular'`, right?

(also, these if statements should be elifs)